Lib/test/_test_multiprocessing.py

```diff
@@ -4440,6 +4440,31 @@ def test_shared_memory_cleaned_after_process_termination(self):
                     "resource_tracker: There appear to be 1 leaked "
                     "shared_memory objects to clean up at shutdown", err)
 
+    @unittest.skipIf(os.name != "posix", "resource_tracker is posix only")
+    def test_shared_memory_untracking(self):
+        # gh-82300: When a separate Python process accesses shared memory
+        # with track=False, it must not register with the resource tracker.
+        cmd = '''if 1:
+            import sys
+            from unittest.mock import Mock
+            from multiprocessing import resource_tracker
+            from multiprocessing.shared_memory import SharedMemory
+            resource_tracker.register = Mock(side_effect=AssertionError)
+            mem = SharedMemory(create=False, name=sys.argv[1], track=False)
+            mem.close()
+        '''
+        mem = shared_memory.SharedMemory(create=True, size=10)
+        try:
+            *_, err = test.support.script_helper.assert_python_ok("-c", cmd,
+                                                                  mem.name)
+            self.assertEqual(err, b'')
+        finally:
+            mem.close()
+            try:
+                mem.unlink()
+            except OSError:
```

Comment:
Is there a situation where `OSError` would be raised here?

Reply:
Good point. The subprocess is never given the chance to register the resource tracker so there is never any risk of the subprocess resource tracker cleaning up. Will remove the try-except.

Reply:
technically you don't know why the subprocess died. it could've started the resource tracker and died for any reason (before or after registering the memory with the tracker)... This is a test suite run in CI on prerelease code... these things happen 😅.

So I think moving the `assert_python_ok` within the `try:` block and accepting that even `resource_tracker.unregister` may fail as well as `mem.unlink` is good defensive state cleanup coding here.

Reply:
Good points. The code should be a bit more defensive on cleanup now. 

Unregistering of the resource tracker should not be allowed to fail. This is the resource tracker process that belongs to the main process.